rust/kernel/chrdev.rs

```diff
@@ -149,12 +197,9 @@ unsafe impl<const N: usize> Sync for Registration<{ N }> {}
 impl<const N: usize> Drop for Registration<{ N }> {
     fn drop(&mut self) {
         if let Some(inner) = self.inner.as_mut() {
-            // SAFETY: Calling unsafe functions, `0..inner.used` of
-            // `inner.cdevs` are initialized in `Registration::register`.
+            // SAFETY: TODO needs explanation why unregister_chrdev_region
+            // is safe here.
             unsafe {
-                for i in 0..inner.used {
-                    bindings::cdev_del(inner.cdevs[i].as_mut_ptr());
-                }
                 bindings::unregister_chrdev_region(inner.dev, N.try_into().unwrap());
```

Comment:
Everywhere in the kernel, `cdev_del` happens before `unregister_chrdev_region`. We do the opposite here, because the `[Option<Cdev>; N]` is dropped only at the end of this `drop` function.

Are we sure we want to go against the grain here, even though we assume this is *probably* ok?

A safe way out could consist of what we had in v1 of this PR:
```rust
for i in 0..inner.used {
  inner.cdevs[i].take();
}
bindings::unregister_chrdev_region(inner.dev, N.try_into().unwrap());
```

Reply:
@ojeda how would you like to proceed with this PR? I'm ok to pause, change approach, or perhaps abort if you feel we're not ready for this change yet.

Reply:
I think it's best to do like everyone else, i.e., `cdev_del` first. (Sorry, my comment was the one that lead to the change from v1.)

Let's proceed with the PR.

Reply:
Agreed, we should not deviate.